### PR TITLE
nixos/prowlarr: add package and datadir options

### PR DESCRIPTION
**Pull request by paroxp**
###### Description of changes

This should bring Prowlarr a little closer to sonarr and radarr in terms of alignment. Mainly, allows choosing a different source for the package than the stable branch.

**The default value for `dataDir` destination changes.**

Users might want to set their old destination path in the config, or manually move the files:

```
mkdir /tmp/Prowlarr
mv -r /var/lib/prowlarr/* /tmp/Prowlarr
mkdir -p /var/lib/prowlarr/.config
mv -r /tmp/Prowlarr /var/lib/prowlarr/.config
```

###### Things done

Didn't get to testing. Based of a different package.

Resolves #195273

### CODE REVIEW

File: nixos/modules/services/misc/prowlarr.nix [hunk 1]
```diff
@@ -11,11 +11,26 @@ in
     services.prowlarr = {
       enable = mkEnableOption (lib.mdDoc "Prowlarr");
 
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/prowlarr/.config/Prowlarr";
```
Comment (NickCao) [inline nixos/modules/services/misc/prowlarr.nix:16]: Why is this different from the original default (/var/lib/prowlarr)

Comment (paroxp) [reply nixos/modules/services/misc/prowlarr.nix:16]: Technically, it doesn't have to. It doesn't even matter... In fact, the official docs also mention `/var/lib/prowlarr` so, I am being weird here.

I have however, attempted to normalise it with the line of `sonarr` and `radarr` for convinience of navigation. For instance:

```
/var/lib/sonarr/.config/NzbDrone/config.xml
/var/lib/radarr/.config/Radarr/config.xml
/var/lib/prowlarr/config.xml
```

I'll be happy to reset the default value to the original if you'd rather not break the user experience.

**Review by mweinelt — CHANGES_REQUESTED** [review]
Please use a bind mount instead, as a separate `dataDir` will not work well with `DynamicUser`´and I'm not in favor of allocating a static uid/gid for prowlarr.

Read up on the discussion in the PR where this module was created: https://github.com/NixOS/nixpkgs/pull/136039#discussion_r722317458

**Comment by paroxp** [comment]
I've had a read through your comment. And whilst I agree with you on the user:group issue, I am not sure how this affects the directory. That is however, my lack of understanding or context.

I've had a go at solving this, first with:

```
    fileSystems."/var/lib/prowlarr" = {
      device = cfg.dataDir;
      options = [ "bind" "rw" ];
      fsType = "none";
    };
```

but that sounded nonsensical to me.  Then deciding you actually mean a property of `systemd` configuration file.

With that in mind, do have a look at my latest commit, see if this is what you meant or need. Do advise otherwise - systemd newbie here.

File: nixos/modules/services/misc/prowlarr.nix [hunk 2]
```diff
@@ -4,18 +4,32 @@ with lib;
...
         description = lib.mdDoc "Open ports in the firewall for the Prowlarr web interface.";
       };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.prowlarr;
+        defaultText = literalExpression "pkgs.prowlarr";
+        description = lib.mdDoc ''
+          Prowlarr package to use.
+        '';
```
Comment (SuperSandro2000) [inline nixos/modules/services/misc/prowlarr.nix:31]: This got already added in the meantime

**Pull request closed without merging**